Classes/Search/SearchViewController.swift

```diff
@@ -0,0 +1,151 @@
+//
+//  SearchViewController.swift
+//  Freetime
+//
+//  Created by Sherlock, James on 28/07/2017.
+//  Copyright © 2017 Ryan Nystrom. All rights reserved.
+//
+
+import UIKit
+import IGListKit
+
+class SearchViewController: UIViewController,
+                            ListAdapterDataSource,
+                            FeedDelegate,
+                            SearchLoadMoreSectionControllerDelegate,
+                            SearchBarSectionControllerDelegate {
+
+    private let client: GithubClient
+    private lazy var feed: Feed = { Feed(viewController: self, delegate: self) }()
+    private var searchResults = [SearchResult]()
+    private var nextPage: String?
+    private var searchTerm: String?
+    
+    private let noResultsKey = "noResultsKey" as ListDiffable
+    private let loadMore = "loadMore" as ListDiffable
+    private let searchKey = "searchKey" as ListDiffable
+    
+    init(client: GithubClient) {
+        self.client = client
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        feed.viewDidLoad()
+        feed.collectionView.refreshControl?.endRefreshing()
+        feed.adapter.dataSource = self
+        title = "Search"
+    }
+    
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+        feed.viewWillLayoutSubviews(view: view)
+    }
+    
+    // MARK: Data Loading/Paging
+    
+    private func update(dismissRefresh: Bool, animated: Bool = true) {
+        feed.finishLoading(dismissRefresh: dismissRefresh, animated: animated)
+    }
+    
+    private func handle(resultType: GithubClient.SearchResultType, append: Bool, animated: Bool) {
+        switch resultType {
+        case .error:
+            print("😞 Something went wrong, here's some pizza 🍕")
```

Comment:
Jokes aside while a better message could probably be shown, it does show the status bar error screen and then an empty results placeholder view so the user isn't left in limbo

Similar experience to notifications view